PR: https://github.com/odoo/odoo/pull/

From: ab303435d3484d1ab5ba87cead7382e2d360399e
From: Fabien Pinckaers
Date: 2015-11-19 14:15:21

Breaking data model changes scores: del:3 + add:2, change matches:
-    activity_1_id = fields.Many2one('crm.activity', string="Next Activity 1")
-    activity_2_id = fields.Many2one('crm.activity', string="Next Activity 2")
-    activity_3_id = fields.Many2one('crm.activity', string="Next Activity 3")
+    activity_ids = fields.Many2many('crm.activity', 'crm_activity_rel', 'activity_id', 'recommended_id', string='Recommended Activities')
+    activity_invert_ids = fields.Many2many('crm.activity', 'crm_activity_rel', 'recommended_id', 'activity_id', string='Preceding Activities')

Total Changes: 6

[IMP] better handling of next activities

================================= pseudo patch: =================================

--- a/addons/crm/models/crm_activity.py
+++ b/addons/crm/models/crm_activity.py
@@ -29,9 +29,9 @@ class CrmActivity(models.Model):
     sequence = fields.Integer('Sequence', default=0)
     team_id = fields.Many2one('crm.team', string='Sales Team')
     subtype_id = fields.Many2one('mail.message.subtype', string='Message Subtype', required=True, ondelete='cascade')
-    activity_1_id = fields.Many2one('crm.activity', string="Next Activity 1")
-    activity_2_id = fields.Many2one('crm.activity', string="Next Activity 2")
-    activity_3_id = fields.Many2one('crm.activity', string="Next Activity 3")
+
+    activity_ids = fields.Many2many('crm.activity', 'crm_activity_rel', 'activity_id', 'recommended_id', string='Recommended Activities')
+    activity_invert_ids = fields.Many2many('crm.activity', 'crm_activity_rel', 'recommended_id', 'activity_id', string='Preceding Activities')
 
     @api.model
     def create(self, values):
